dspace-api/src/main/java/org/dspace/subscriptions/ContentGenerator.java

```diff
@@ -0,0 +1,96 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.dspace.content.Item;
+import org.dspace.content.crosswalk.StreamDisseminationCrosswalk;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.core.Email;
+import org.dspace.core.I18nUtil;
+import org.dspace.discovery.IndexableObject;
+import org.dspace.eperson.EPerson;
+import org.dspace.subscriptions.service.SubscriptionGenerator;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * Implementation class of SubscriptionGenerator
+ * which will handle the logic of sending the emails
+ * in case of 'content' subscriptionType
+ */
+@SuppressWarnings("rawtypes")
+public class ContentGenerator implements SubscriptionGenerator<IndexableObject> {
+
+    private final Logger log = LogManager.getLogger(ContentGenerator.class);
+
+    @SuppressWarnings("unchecked")
+    private Map<String, StreamDisseminationCrosswalk> entityType2Disseminator = new HashMap();
+
+    @Autowired
+    private ItemService itemService;
+
+    @Override
+    public void notifyForSubscriptions(Context context, EPerson ePerson,
+                                       List<IndexableObject> indexableComm,
+                                       List<IndexableObject> indexableColl) {
+        try {
+            if (Objects.nonNull(ePerson)) {
+                Locale supportedLocale = I18nUtil.getEPersonLocale(ePerson);
+                Email email = Email.getEmail(I18nUtil.getEmailFilename(supportedLocale, "subscriptions_content"));
+                email.addRecipient(ePerson.getEmail());
+                email.addArgument(generateBodyMail(context, indexableComm));
+                email.addArgument(generateBodyMail(context, indexableColl));
+                email.send();
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+            log.warn("Cannot email user eperson_id: {} eperson_email: {}", ePerson::getID, ePerson::getEmail);
+        }
+    }
+
+    private String generateBodyMail(Context context, List<IndexableObject> indexableObjects) {
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            out.write("\n".getBytes(UTF_8));
+            if (indexableObjects.size() > 0) {
+                for (IndexableObject indexableObject : indexableObjects) {
+                    out.write("\n".getBytes(UTF_8));
+                    Item item = (Item) indexableObject.getIndexedObject();
+                    String entityType = StringUtils.defaultIfBlank(itemService.getEntityTypeLabel(item),
+                                                                   "Item");
+                    entityType2Disseminator.get(entityType).disseminate(context, item, out);
```

Comment:
@corrad82-4s and @Micheleboychuk : I feel I should point out that this line will *still* result in a NullPointerException if a site adds a custom EntityType and *forgets* to add it to the list of entities in `subscriptions_email_configuration.xml`.

That seems like bad behavior if we want to support future Entities (it also will likely be a problem for DSpace-CRIS when you attempt to update DSpace-CRIS to use this code, as you'll hit NullPointerExceptions until all your custom entities are correctly listed).

Wouldn't it be much easier (and less error prone) to do something like this...
```
StreamDisseminationCrosswalk crosswalk = entityType2Disseminator.get(entityType);
// If crosswalk cannot be found, fallback to "Item" crosswalk
if (crosswalk == null) {
  crosswalk =  entityType2Disseminator.get("Item");
}
crosswalk.disseminate(context, item, out);
```

Reply:
Hi @tdonohue , this feedback has been addressed